src/cata_tiles.cpp

```diff
@@ -1,69 +1,373 @@
 #if (defined SDLTILES)
 #include "cata_tiles.h"
+#include "game.h"
+#include "options.h"
+#include "mapdata.h"
 #include "debug.h"
 #include "json.h"
 #include "path_info.h"
 #include "monstergenerator.h"
 #include "item.h"
+#include "item_factory.h"
 #include "veh_type.h"
 #include "filesystem.h"
 #include "sounds.h"
 
 #include <algorithm>
 #include <fstream>
+#include <memory>
 
+#include "SDL2/SDL.h"
+#include "SDL2/SDL_ttf.h"
 #include "SDL2/SDL_image.h"
 
 #define dbg(x) DebugLog((DebugLevel)(x),D_SDL) << __FILE__ << ":" << __LINE__ << ": "
 
-#define ITEM_HIGHLIGHT "highlight_item"
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// template specializations inside std are explicitly allowed by the standard
+////////////////////////////////////////////////////////////////////////////////////////////////////
+namespace std {
+template<> struct default_delete<SDL_Surface> {
+    void operator()(SDL_Surface* const ptr) const noexcept { SDL_FreeSurface(ptr); }
+};
+template<> struct default_delete<SDL_Texture> {
+    void operator()(SDL_Texture* const ptr) const noexcept { SDL_DestroyTexture(ptr); }
+};
+} //namespace std
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////
+enum class light_type : int {
+    clear,
+    light_normal,
+    light_dark,
+    boomer_normal,
+    boomer_dark,
+    hidden = -1,
+};
 
-extern game *g;
-//extern SDL_Surface *screen;
-extern int WindowHeight, WindowWidth;
-extern int fontwidth, fontheight;
+// Make sure to change to_string if this changes!
+enum class tile_category : int {
+    none,
+    vehicle_part,
+    terrain,
+    item,
+    furniture,
+    trap,
+    field,
+    lighting,
+    monster,
+    bullet,
+    hit_entity,
+    weather,
+    num_tile_categories
+};
+
+std::string const& to_string(tile_category const cat)
+{
+    constexpr int size = static_cast<int>(tile_category::num_tile_categories);
+    static std::array<std::string, size> const strings {{
+        "",
+        "vehicle_part",
+        "terrain",
+        "item",
+        "furniture",
+        "trap",
+        "field",
+        "lighting",
+        "monster",
+        "bullet",
+        "hit_entity",
+        "weather",
+    }};
+
+    auto const i = static_cast<int>(cat);
+    if (i < 0 || i >= size) {
+        dbg(D_ERROR) << "bad tile_category value";
+        strings[0];
```

Comment:
@kevingranade thanks, I'll test again with that tileset after I get up tomorrow, but I hope the change I made was enough.